frontend/src/components/modals/NotifyUsersModal.tsx

```diff
@@ -0,0 +1,223 @@
+    [],
+  );
+  const [previousUsers, setPreviousUsers] = useState<
+    CheckableUser[] | undefined
+  >([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [allChecked, setAllChecked] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (
+      getType(userInfo?.roles, task.roadmapId) === RoleType.Admin &&
+      customers &&
+      allUsers
+    ) {
+      const missingCustomers = findMissingCustomers(task.ratings, customers);
+      const missingRepresentatives = new Map<number, CheckableUser>();
+
+      // Find representatives who haven't given ratings
+      if (missingCustomers) {
+        missingCustomers.forEach((customer) => {
+          // Filter out logged in user, so they won't send email to themselves
+          const representatives = customer?.representatives?.filter(
+            (user) => user.id !== userInfo?.id,
+          );
+          const ratingsForTask = task.ratings
+            .filter((rating) => rating.forCustomer === customer.id)
+            .map((e) => e.createdByUser);
+
+          if (representatives) {
+            representatives.forEach((rep) => {
+              if (!ratingsForTask.includes(rep.id))
+                if (
+                  !Array.from(missingRepresentatives.values()).some(
+                    (e) => e.id === rep.id,
+                  )
+                )
+                  missingRepresentatives.set(rep.id, {
+                    ...rep,
+                    checked: false,
+                  });
+            });
+          }
+        });
+      }
+
+      const missingDevelopers = findMissingDevelopers(
+        task.ratings,
+        allUsers,
+      ).map((dev) => ({ ...dev, checked: false }));
+
+      setMissingUsers([
+        ...(Array.from(missingRepresentatives.values()) ?? []),
+        ...(missingDevelopers ?? []),
+      ]);
+    }
+  }, [allUsers, customers, task.ratings, task.roadmapId, userInfo]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsLoading(true);
+
+    const usersToNotify = missingUsers
+      ?.filter((user) => user.checked)
+      .map((user) => user.id);
+
+    // Messagefield needs design
+    if (usersToNotify) {
+      const res = await dispatch(
+        roadmapsActions.notifyUsers({
+          users: usersToNotify,
+          task,
+          message: 'test',
+        }),
+      );
+      setIsLoading(false);
+      if (roadmapsActions.notifyUsers.rejected.match(res)) {
+        if (res.payload) setErrorMessage(res.payload.message);
+        return;
+      }
+      closeModal();
+    }
+  };
+
+  const checkAll = (checked: boolean) => {
+    if (!missingUsers) return;
+    if (!checked) setMissingUsers(previousUsers);
+    if (checked) {
+      const copy = [...missingUsers];
+      setPreviousUsers(copy);
+      setMissingUsers(copy.map((_user) => ({ ..._user, checked })));
```

Comment:
The underscore should be dropped, as it usually indicates unused parameter
```suggestion
      setMissingUsers(copy.map((user) => ({ ...user, checked })));
```